pkg/gofr/http/responder.go

```diff
@@ -35,9 +36,11 @@ func (r Responder) Respond(data interface{}, err error) {
 
 		return
 	default:
-		resp = response{
-			Data:  v,
-			Error: errorObj,
+		resp = response{Data: data, Error: errorObj}
+
+		// handling where an interface contains a nullable type with a nil value.
+		if isNil(data) {
+			resp = response{Error: errorObj}
 		}
```

Comment:
The code is a bit strange to read

I would expect something like


```suggestion
		resp = response{Data: data, Error: errorObj}

		// handling where an interface contains a nullable type with a nil value.
		if isNil(data) {
			resp.Data = nil
 		}
```

or


```suggestion
		resp = response{Error: errorObj}

		if !isNil(data) {
			resp.Data = data
 		}
```

Reply:
@ccoVeille  sure we can go with 			
```go
resp.Data = nil
```

Reply:
Maybe I wasn't able to explain correctly.

I'll suggest something else